packages/core/src/Site/index.js

```diff
@@ -844,9 +851,39 @@ class Site {
   }
 
   async reloadSiteConfig() {
+    const oldSiteConfig = this.siteConfig;
     const oldAddressablePages = this.addressablePages.slice();
     const oldPagesSrc = oldAddressablePages.map(page => page.src);
     await this.readSiteConfig();
+    await this.handleIgnoreReload(oldSiteConfig);
+    await this.handlePageReload(oldAddressablePages, oldPagesSrc,
+                                oldSiteConfig, this.isGlobalConfigModified(oldSiteConfig));
```

Comment:
hmm should we `isGlobalConfigModified` inside `handlePageReload`? since its not used anywhere else and concerns page rebuilding only

Reply:
sorry, actually just meant changing the call site. But the current changes are ok as well : )